### PR TITLE
Extend cholesky of triangular dot rewrite to matmul Ops

### PR DESCRIPTION
**Pull request by ricardoV94**
Also restrict to 2D Dot cases

Also reverts #455

### CODE REVIEW

File: pytensor/tensor/rewriting/linalg.py [hunk 1]
```diff
@@ -168,13 +168,25 @@ def cholesky_ldotlt(fgraph, node):
...
-    if not (A.owner and isinstance(A.owner.op, (Dot, Dot22))):
+    if not (
+        A.owner is not None
+        and (
+            (
+                isinstance(A.owner.op, (Dot, Dot22))
+                # This rewrite only applies to matrix Dot
+                and A.owner.inputs[0].type.ndim == 2
+            )
+            or (A.owner.op == _matrix_matrix_matmul)
```
Comment (jessegrabowski) [inline pytensor/tensor/rewriting/linalg.py:187]: So I understand -- this is saying the rewrite applies to: 1) normal 2d matrix multiplication (Dot, Dot22), or 2) a blockwise matrix multiplication (`_matrix_matrix_matmul`)? 

Pretty amazing that no other special logic is needed to handle the blockwise simplification!

File: tests/tensor/rewriting/test_linalg.py [hunk 2]
```diff
@@ -137,26 +140,30 @@ def test_matrix_inverse_solve():
...
 @pytest.mark.parametrize("cholesky_form", ("lower", "upper"))
 @pytest.mark.parametrize("product", ("lower", "upper", None))
-def test_cholesky_ldotlt(tag, cholesky_form, product):
+@pytest.mark.parametrize("op", (dot, matmul))
+def test_cholesky_ldotlt(tag, cholesky_form, product, op):
     transform_removes_chol = tag is not None and product == tag
     transform_transposes = transform_removes_chol and cholesky_form != tag
 
-    A = matrix("L")
+    ndim = 2 if op == dot else 2
```
Comment (jessegrabowski) [inline tests/tensor/rewriting/test_linalg.py:148]: else 3? This line doesn't seem to do anything

**Review by jessegrabowski — COMMENTED** [review]
Looks good pending that you check the typo on `ndim` in the unittest

File: tests/tensor/rewriting/test_linalg.py (hunk 2 above)
Comment (ricardoV94) [reply tests/tensor/rewriting/test_linalg.py:148]: Good catch. I rewrote the test too many times

**Review by jessegrabowski — APPROVED** [review]
LGTM

**Pull request merged**